assets/src/components/detours/detourModal.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react"
+import { DiversionPage } from "./diversionPage"
+import { StartDetourProps } from "./detourDropdown"
+
+export const DetourModal = ({
```

Comment:
This is cool and all, but I wonder if it can be combined with `DiversionPage`...

But maybe not, because this does take care of the modal logic itself, which... actually might be worth keeping separate.

Thoughts?

Reply:
I'd prefer to compose them instead of collapsing them.

Reply:
@firestack are you thinking something like 👇?

```typescript
<DetourModal show={showDetourModal} onClose={() => {doTheRightThing()}}>
  <DiversionPage {...allTheProps} />
</DetourModal>
```

Reply:
I think what I was trying to convey is that, similarly to how the Mobile Nav Menu is somewhat annoying to use (and display in storybook) because it handles both displaying itself on the screen, positioning, AND being a nav, I like that `DetourModal` and `(Detour/Diversion)Page` are separate and that the page doesn't concern itself with creating it's parent container and positioning it (though I'll admit it's breaking some encapsulation boundaries by having the close button be in the Page and not the modal, but going that direction is maybe too "pure")

The example you show is certainly one way to achieve that, but you could also let the modal do the work of figuring out `allTheProps` and "hardcode" the `DiversionPage` as the `DetourModal`'s only child for now. so the example would be more like
```
<DetourModal
	show={showDetourModal}
	onClose={() => {doTheRightThing()}}
	routePatternId={...}
	{any other id's needed to derive information for the `DiversionPage`}
/>
```

I'm assuming that "hard-coding" it would be simpler/easier for now.

But if the number of props that `DetourModal` needs to prop-drill, without modification, 
becomes large enough that we're thinking of putting them in a struct to move it, then I think it'd be better to do via composition, like your [example above](https://github.com/mbta/skate/pull/2377#discussion_r1490110281).

Reply:
I think I also prefer the idea of composition, and using `grid` and `subgrid` (like `DiversionPage` and `DiversionPanel` do), I think it might be possible to get the close button to be fully a part of the `DetourModal` without the `DiversionPage` knowing about it at all.

I would be interested in trying that out, although I might choose to do that as a follow-up instead.

Reply:
Heh, using `grid` and `subgrid` to get the button to belong to the modal but appear on the diversion page was exactly what I was thinking. If it's easy, than yay! but otherwise it's _probably_ not worth our effort at this time?

Reply:
I'll prioritize this last so that the PR can get merged without nice composition, and then follow up with nice composition later on if I can figure it out.

(Or I'll sneak it into this PR if I _can_ figure it out reasonably quickly)